packages/structured-clone/vite.config.ts

```diff
@@ -1,7 +1,7 @@
 import { LibraryOptions, defineConfig } from "vite";
 import dts from "vite-plugin-dts";
 import { isCjsPackage } from "../../scripts/esbuild-module-check.mjs";
-import { packageReadmeAndPackageJson } from "../../scripts/vite-plugins";
+import { packageReadmeAndPackageJson } from "../../scripts/vite-plugins.js";
```

Comment:
Did this import fail without the extension? I would think `vite` would be able to find it without trouble...

Reply:
This is one of the things I don't quite understand. It certainly seems to work without the extension.
However, the document of vite says:

> The vite.config.js file content is using the ESM syntax.

The ES module does not allow imports without extensions. At least nodejs does.

https://nodejs.org/api/esm.html#import-specifiers

> Relative specifiers like './startup.js' or '../config.mjs'. They refer to a path relative to the location of the importing file. The file extension is always necessary for these.

Reply:
Since this is a `.ts` file, I don't think we need to follow the `.js` rules.

Reply:
In practice, this rule remains the same for both `.ts` and `.js`. To ease the discomfort of importing `.ts` files as `.js`, TypeScript 5.0 or later allows importing in `.ts`.
https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#allowimportingtsextensions
（However, typescript does not change the extension in the import at compile time 😢）

Import without extension is called fake ESM (or faux ESM). I think `vite.config.ts` is probably working well complemented by vite.
Code written in the fake ESM will not work in the native ESM. This is as I mentioned in the #72 .
We can write imports in `vite.config.ts` without the extension, but I am not sure if this will work in the future.

Reply:
It will continue to work. `vite`'s policy is to try to be easy to use, and I doubt they'll change that.

Please remove the `.js` from the imports that were changed.

Reply:
OK. I dropped the changes.